src/wallet/walletdb.h

```diff
@@ -58,13 +61,16 @@ class CHDChain
     {
         READWRITE(this->nVersion);
         READWRITE(nExternalChainCounter);
+        if (this->nVersion >= VERSION_HD_CHAIN_SPLIT)
```

Comment:
This should set nInternalChainCounter for old versions - maybe use `std::numeric_limits<uint32_t>::max()` (and make it private, with all accessors throwing if it's not a sufficient version)?

Also, braces, please.